### DISCUSSION TITLE
Better integration with VSCode

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [devspace-sh/devspace#2187] [Ideas]
**Is your feature request related to a problem?**  

Let's say I was looking to use start a Devspace dev environment on my low power MBA (well it is a few years ago), so I can work it out on the go. Right now you just setup SSH and then use the VSCode remote extension which is being quite obviously. However I don't know how to keep VSCode Remote state and cache the dev instance and I expected to lose extensions on the moment I tear down the dev instance. It will also have to reconfigure so it's quite painful.

**Which solution do you suggest?**  

Just give a more comprehensive guide on this matter, or even propose on making an extension for this, say like Gitpod and more recently GitHub Codespace, they both allowed to start the dev container with a custom additive Dockerfile/Nix derivative, or how to upload the dotfiles (Like individual Git username and email and stuff).

**Which alternative solutions exist?**  

Okteto. They offered an extension exactly like what I described. However I don't like them being too anal about their Okteto Cloud product, and so I still have to turn around.

**Additional Context**
I have experimented with VSCode Remote server, and I was greeted with this:
```
> Waiting for server log...
>  
> *
> * Reminder: You may only use this software with Visual Studio family products,  
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057) 
> *
> 
> Server did not start successfully. Full server log at /root/.vscode-server/.c351
> 11e6c69bb39013c4a4b7b9566ec1ca73fc4d5.log >>>
> /root/.vscode-server/bin/c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5/bin/code-serve
> er: line 12: /root/.vscode-server/bin/c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5/no
> ode: not found
> <<< End of server log
> 06481c347845: start
```

I reached out to the PWD of the node executable, and it is obviously something wrong:
```
bash-5.1# /root/.vscode-server/bin/c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5/node
bash: /root/.vscode-server/bin/c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5/node: No such file or directory
bash-5.1# cd /root/.vscode-server/bin/c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5/
bash-5.1# ls -lah
total 76M
drwxr-xr-x    6 root     root        4.0K Jul 25 13:15 .
drwxr-xr-x    3 root     root        4.0K Jul 25 13:14 ..
-rw-rw-r--    1 root     root       13.1K May 17 18:23 LICENSE
drwxr-xr-x    4 root     root        4.0K Jul 25 13:14 bin
drwxr-xr-x   33 root     root        4.0K Jul 25 13:14 extensions
-rwxr-xr-x    1 root     root       75.6M May 17 18:24 node
drwxr-xr-x  166 root     root        4.0K Jul 25 13:14 node_modules
drwxr-xr-x    3 root     root        4.0K Jul 25 13:14 out
-rw-rw-r--    1 root     root          62 May 17 18:23 package.json
-rw-rw-r--    1 root     root       36.6K May 17 18:24 product.json
-rwxr-xr-x    1 root     root         240 May 17 18:22 server.sh
```

Since I'm running Alpine, I tried my luck on gcompact and friends
```
bash-5.1# apk add gcompat libgcc libstdc++
```
And it seems like it worked!
```
bash-5.1# ./node -h
Usage: node [options] [ script.js ] [arguments]
       node inspect [options] [ script.js | host:port ] [arguments]
```
I'm not sure if anything would crash, but I would say this is the crucial first step.

**Comment by FabianKramm** [comment]
Hey @stevefan1999-personal ! Sorry for hanging this so long, but we wrote a guide on how to use VSCode with DevSpace: https://devspace.sh/docs/ide-integration/visual-studio-code . If losing extensions and data is a problem after restarts, you can use the persist paths option to persist certain folders in a persistent volume: https://devspace.sh/docs/configuration/dev/modifications/persistence .
